<launch>

	<!-- replay bag data -->
	<node pkg="rosbag" type="play" name="rosbag_play" args="$(find ugv_localization)/bag/test_02.bag"/>


	<!-- combine imu topics and convert to convention -->
	<node pkg="ugv_localization" type="imu_combine.py" name="imu_combine" output="screen"/>


	<!-- get 2D baseline odometry using wheel encoder and imu -->
	<node pkg="ugv_localization" type="wheel_imu_odom.py" name="wheel_imu_odom" args="/wheel_odometry /imu/combined /odometry/wheel_imu" output="screen"/>


	<!-- fusion by robot_localization package -->
 	<node pkg="robot_localization" type="ekf_localization_node" name="ekf_odom_imu" clear_params="true">
		<rosparam command="load" file="$(find ugv_localization)/config/global/ekf_odom_imu.yaml"/>
		<remap from="odometry/filtered" to="odometry/intermediate"/>
	</node>

    <node pkg="robot_localization" type="navsat_transform_node" name="navsat_transform_node" respawn="true" clear_params="true">
    	<rosparam command="load" file="$(find ugv_localization)/config/global/navsat_transform.yaml"/>
    
    	<remap from="imu/data" to="imu/combined"/>
    	<remap from="gps/fix" to="rtk_ublox/fix"/>
    	<remap from="odometry/filtered" to="odometry/intermediate"/>
 	</node>

	<node pkg="ugv_localization" type="global_filter_wrapper.py" name="global_filter_wrapper" args="/odometry/wheel_imu /odometry/gps /imu/combined" output="screen">
		<param name="config_file" value="$(find ugv_localization)/config/global/ekf_odom_imu_gps.yaml"/>
	</node>


    <!-- publish required tf (transformation) -->
	<node pkg="tf2_ros" type="static_transform_publisher" name="base_link2gps" args="0 0 0 0 0 0 base_link gps"/>


	<!-- plot in python -->
    <!--node pkg="ugv_localization" type="plot_traj.py" name="plot_traj_odom" args="/odometry/wheel_imu odom" output="screen"/-->

    <!--node pkg="ugv_localization" type="plot_traj.py" name="plot_traj_imu" args="/imu_3dm/imu imu" output="screen"/-->

	<!--node pkg="ugv_localization" type="plot_traj.py" name="plot_traj_gps" args="/rtk_ublox/fix gps" output="screen"/-->
	

	<!-- visualize in rviz -->
    <node pkg="rviz" type="rviz" name="rviz" args="-d $(find ugv_localization)/rviz/global.rviz"/>

</launch>
